talk/basicconcepts/headersinterfaces.tex

```diff
@@ -73,3 +73,51 @@
     \end{cppcode*}
   \end{block}
 \end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Header / source separation}
+  \begin{goodpractice}[Header / source]{Headers and source files}
+    \begin{itemize}
+      \item Headers should only contain declarations that are necessary to know the interface of functions / classes
+      \item Headers might be included and compiled many times, so it is good to keep them short
+      \begin{itemize}
+        \item The same goes for \cppinline{#include} statements in headers
+      \end{itemize}
+      \item Implementation files only get compiled once, so put necessary includes and most of the code here
+        \begin{itemize}
+          \item Exception: Short functions can be in headers
+          \item Templates (see later) should be in headers
+        \end{itemize}
+    \end{itemize}
+  \end{goodpractice}
+  \begin{block}{Header: Feature.hpp}
+    \begin{cppcode*}{}
+    #include "TypeDeclarations.hpp"
+    void feature(Type1 first, Type2 second);
+    \end{cppcode*}
```

Comment:
Random thought: if you pass by const reference, a forward declaration would suffice:
```latex
    \begin{cppcode*}{}
    class Type1;
    class Type2;
    void feature(const Type1& first, const Type2& second);
    \end{cppcode*}
```
This is sometimes forgotton, but may also not be most important thing to teach. I leave it up to you whether to mention this.

Reply:
Remember we are in the basic part here. I would not enter this debate, especially that there is not even a consensus on the best practice here.